reactor-core/src/main/java/reactor/core/publisher/FluxBuffer.java

```diff
@@ -153,9 +153,10 @@ public void onNext(T t) {
 				buffer = b;
 			}
 
-			b.add(t);
-
-			if (b.size() == size) {
+			if (!b.add(t)) {
```

Comment:
Will do; I originally wrote it this way to avoid a nested if for the positive case, but I can see the convention elsewhere is to test the positive first, then nest if necessary